docs/why-lakefs.md

```diff
@@ -0,0 +1,107 @@
+---
+layout: default
+title: Why is lakeFS the data solution you’ve been missing?
```

Comment:
Why is this a top level element? This looks strange:
![image](https://user-images.githubusercontent.com/3149885/173556507-f2dd6e3b-2e09-40a4-9aaa-1a8ee7563ed2.png)

Reply:
The "what" and the "why" is the most important things you want to know about any one topic. They should be really easy to find in a doc, either as standalone elements or inside of the same doc.

Personally I think it's important to think about snackable content, not too long articles, that's the reason why I would raise this PR, I felt the existing article was a bit compact.

I see your point on rendering but what about this approach?

lakeFS/
  Why
  What

@adipolak wdyt?